### PR TITLE
Add a warning if the debugger is not signed and running on OS X

### PR DESCRIPTION
**Pull request by edumunoz**
Previously at https://github.com/Microsoft/MIEngine/pull/328

### CODE REVIEW

**Comment by edumunoz** [comment]
@gregg-miskelly suggested the verification might be expensive (hashing the file, etc.).

We could go the route of dumping the signing data (`codesign -d`), which won't verify it's correct but at least it verifies it has been signed somehow.

@gregg-miskelly, do you think that may be good enough?

**Comment by gregg-miskelly** [comment]
Sounds reasonable to me. But this is your call. I only wanted to bring up the issue because we have had problems with this type of thing on Windows in the past.

**Comment by edumunoz** [comment]
Yeah, I'll check with the rest to see what everyone thinks.

Ping @andrewcrawley @jacdavis

**Comment by edumunoz** [comment]
Ping @pieandcakes too

**Comment by jacdavis** [comment]
:shipit:

**Comment by andrewcrawley** [comment]
LGTM.

File: src/MIDebugEngine/ResourceStrings.resx [hunk 1]
```diff
@@ -207,4 +207,8 @@
   <data name="Warning_UsingDefaultArchitecture" xml:space="preserve">
     <value>Warning: Debuggee TargetArchitecture not detected, assuming x86_64.</value>
   </data>
+  <data name="Warning_DarwinDebuggerUnsigned" xml:space="preserve">
+    <value>Debugger executable '{0}' is not signed. Debugging may not work.</value>
```
Comment (pieandcakes) [inline src/MIDebugEngine/ResourceStrings.resx]: "Debugging may not work properly." or "Debugging may not work as expected."

**Comment by pieandcakes** [comment]
I think if -d just verifies that its signed (not that its valid) I would call that good enough instead of doing the -v full verification against valid CA and revocation lists.

**Pull request merged**